src/src/Helpers/StringParser.cpp

```diff
@@ -26,6 +26,20 @@
 /********************************************************************************************\
    Parse string template
  \*********************************************************************************************/
+// Checks str for the given escaped characters
+// So far \\% \\[ and \\] are used (all with single backslash!)
+bool hasEscapeCharacters(String *str, const String EscapeChar)

```

Comment:
Another, little more generic, remark on this function definition.
You are using explicit pointer definition(s) for the String at hand, and though that's not _wrong_, using a reference (`&`) is more efficient, especially in an embedded environment. References are fairly often used in ESPEasy, also in favor of creating a new pointer that gets passed via the stack, and removed afterward, instead of only the reference(-pointer).
I've seen this in other parts of your code too, and I don't want to start a discussion on it, it's more a thing I noticed, and wondered about. I assume it's the C-programming background you have been using for (quite) some time, but as said, for embedded programming, references are a somewhat better choice. I do appreciate the more explicit use of the `&` pointer-notation when passing the argument, but efficiency gets a higher score for me 😅

Reply:
One good argument to use references (`&`) instead of pointers is that you put the responsibility to the caller to make sure it a valid constructed object and no need to check for null pointers.
Less checks mean smaller code and it is more predictable.

Reply:
> Another, little more generic, remark on this function definition. You are using explicit pointer definition(s) for the String at hand, and though that's not _wrong_, using a reference (`&`) is more efficient, especially in an embedded environment. References are fairly often used in ESPEasy, also in favor of creating a new pointer that gets passed via the stack, and removed afterward, instead of only the reference(-pointer). I've seen this in other parts of your code too, and I don't want to start a discussion on it, it's more a thing I noticed, and wondered about. I assume it's the C-programming background you have been using for (quite) some time, but as said, for embedded programming, references are a somewhat better choice. I do appreciate the more explicit use of the `&` pointer-notation when passing the argument, but efficiency gets a higher score for me 😅

Thanks for the detailed information about `byref` parameters.
I'll keep it in mind for further PRs.